### PR TITLE
Add dev certs commands

### PR DESCRIPTION
**Pull request by carabasdaniel**

### CODE REVIEW

File: pkg/cli/cmd/certs.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+)
+
+type CertsCmd struct {
+	List     ListCertsCmd     `cmd:"" help:"list dev certs"`
+	Generate GenerateCertsCmd `cmd:"" help:"generate dev certs"`
+	Trust    TrustCertsCmd    `cmd:"" help:"trust dev certs"`
+}
+
+const (
+	DefaultCertsDir = "~/.config/topaz/certs"
```
Comment (gertd) [inline pkg/cli/cmd/certs.go:25]: How does this resolve on Windows? 

We had to fix a similar bug in the default policy root path, (https://github.com/aserto-dev/topaz/commit/22ea3258fb6ed418df6de3433efc193b2ad9027f) 

`~` expands when your terminal is either powershell or  pwsh but not when using cmd.exe

File: pkg/cli/cmd/certs.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+		if time.Until(certDetails[k].NotAfter) < 0 {
+			isValid = false
+		}
+		table.WithTableRow(k, certDetails[k].NotBefore.Format(time.RFC3339), certDetails[k].NotAfter.Format(time.RFC3339), fmt.Sprintf("%t", isValid))
+	}
+	table.Do()
+	return nil
+}
+
+type GenerateCertsCmd struct {
```
Comment (gertd) [inline pkg/cli/cmd/certs.go:89]: We provide `list`, `generate` and `trust` 

How does one :
* remove certs
* remove trust
* regenerate certs when certs are expired

File: pkg/cli/cmd/cli.go [hunk 3]
```diff
@@ -24,6 +24,7 @@ type CLI struct {
 	Manifest  ManifestCmd  `cmd:"" help:"manifest commands"`
 	Test      TestCmd      `cmd:"" help:"test assertions commands"`
 	Templates TemplateCmd  `cmd:"" help:"template commands"`
+	Certs     CertsCmd     `cmd:"" help:"cert commands"`
```
Comment (gertd) [inline pkg/cli/cmd/cli.go:27]: The commands are ordered in the most common activities first, the least common/frequent ones last, and grouped by commonality (import/export, backup/restore, load/save), as such I would propose the `Certs` entry comes after `Configure`.

File: pkg/cli/certs/trust_windows.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/magefile/mage/sh"
+	"github.com/pkg/errors"
+)
+
+const certStore = `Cert:\CurrentUser\Root`
+
+func AddTrustedCert(certPath string) error {
+	importCmd := fmt.Sprintf("import-certificate -Filepath %s -CertStoreLocation %s", certPath, certStore)
+
+	if err := sh.RunV("powershell", "-Command", importCmd); err != nil {
```
Comment (gertd) [inline pkg/cli/certs/trust_windows.go:15]: After PowerShell 5.1+ the executable name is `pwsh.exe` 

https://learn.microsoft.com/en-us/powershell/scripting/whats-new/differences-from-windows-powershell?view=powershell-7.4#powershell-executable-changes

File: pkg/cli/cmd/certs.go [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+const (
+	DefaultCertsDir = "~/.config/topaz/certs"
+
+	gatewayCertFileName = "gateway"
+	grpcCertFileName    = "grpc"
+	certCommonName      = "topaz"
+)
+
+type ListCertsCmd struct {
+	CertsDir string `arg:"" required:"false" default:"~/.config/topaz/certs" help:"path to dev certs folder" `
```
Comment (gertd) [inline pkg/cli/cmd/certs.go:33]: The UX feels asymmetric, generate requires named argument `--certs-dir` while `list` accepts an positional argument.

```
❯ ./dist/topaz_darwin_arm64/topaz certs generate --certs-dir="~/.config/topaz/certs2"

Generating gateway and grpc dev certs

Please wait .....
Done
❯ ./dist/topaz_darwin_arm64/topaz certs list --certs-dir="~/.config/topaz/certs2"
Usage: topaz certs list [<certs-dir>]

list dev certs

Arguments:
  [<certs-dir>]    path to dev certs folder

Flags:
  -h, --help        Show context-sensitive help.
  -N, --no-check    disable local container status check ($TOPAZ_NO_CHECK)

topaz: error: unknown flag --certs-dir
```

I would expect that:
`topaz certs list` lists the default directory
`topaz certs list --certs-dir="~/.config/topaz/certs2"` and we follow the same pattern and naming `generate` uses when specifying a different cert directory.

**Review by gertd — CHANGES_REQUESTED** [review]
Please see comments

File: pkg/cli/certs/trust_windows.go (hunk 4 above)
Comment (carabasdaniel) [reply pkg/cli/certs/trust_windows.go:15]: image

I'll make a couple of changes here so it works as expected on newer versions also

File: pkg/cli/cmd/certs.go (hunk 1 above)
Comment (carabasdaniel) [reply pkg/cli/cmd/certs.go:26]: That is just a string defined as a constant for default cert path value (not used to resolve paths), in code we use the os.UserHomeDir() call to establish the home directory.

**Review by gertd — APPROVED** [review]

**Pull request merged**